File: typescript/packages/management/src/tools/list_runs.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const listRuns: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const slug = String(input.agent_slug);
  const limit = typeof input.limit === 'number' ? input.limit : undefined;
  return ctx.client.agents.listRuns(ctx.sandboxId, slug, { limit });
};

File: python/copass-management/src/copass_management/tools/list_runs.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def list_runs(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    limit = input.get("limit")
    return await ctx.client.agents.list_runs(
        ctx.sandbox_id,
        str(input["agent_slug"]),
        limit=int(limit) if isinstance(limit, int) else None,
    )
